enhancements/platforms/aws/customer-provisioned-vpc-and-subnets.md

```diff
@@ -0,0 +1,240 @@
+- "@abhinavdahiya"
+- "@sdodson"
+- "@crawford"
+creation-date: 2019-10-10
+last-updated: 2019-10-10
+status: implementable
+replaces:
+  - "https://docs.google.com/document/d/12Nu_OvcNnItD4vaH6G0ky0IeAxnvDSpzp5sVrZv7-8M/"
+---
+
+# customer-provisioned-vpc-and-subnets
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance,
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this?
+
+## Summary
+
+In many large organizations, IT groups are separated into teams. A common
+separation of duties lies between networking/WAN and application or enabling
+technology deployments. One group will provision or assign AWS accounts with
+core management technology. A second group will be responsible for provisioning
+VPC networks complete with subnet & zone selection, routing, VPN connections,
+and ingress/egress rules from the company. Lastly, an applications team (or
+individual business units) are permitted into the account to provision
+application-specific items (instances, buckets, databases, pipelines, load
+balancers, etc.)
+
+To accomplish this, our infrastructure needs and AWS account access can be
+separated into "infrastructure" versus "application". Infrastructure includes
+elements related to the VPC and the networking core components within the VPC
+(VPC, subnets, routing tables, internet gateways, NAT, VPN). Application items
+include things directly touching nodes within the cluster (ELBs, security
+groups, S3 buckets, nodes). This is a typical separation of responsibilities
+among IT groups and deployment/provisioning pipelines within large companies.
+
+## Motivation
+
+### Goals
+
+As an administrator, I would like to create or reuse my own VPCs, subnets and
+routing schemes that I can deploy OpenShift to.
+
+### Non-Goals
+
+There are no changes in expectations regarding publicly addressable parts of the
+cluster.
+
+## Proposal
+
+The installer should be able to install a cluster in an pre-existing VPC.  This
+requires providing subnets too as there is no sane way installer can carve out
+network ranges for the cluster.
+
+There is an expectation that since the VPC and subnets are being shared, the
+installer cannot modify the networking setup (i.e. the route tables for the
+subnets or the VPC options like DHCP etc.).  Therefore the installer can only
+validate the assumptions about the networking setup.  The installer may assume
+that use has NAT gateways, internet gateways, etc. setup, and installation may
+fail if these assumptions are violated.  Or the installer may attempt to
```

Comment:
I think we can get rid of "Or the installer.." here?

Reply:
we're going planning to do at least some network validation.